File: BACKEND/src/main/java/com/ROOMIFY/Roomify/model/ChatMessage.java
package com.ROOMIFY.Roomify.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.time.LocalDateTime;

@Entity
@Table(name = "chat_messages")
@Getter
@Setter
public class ChatMessage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long senderId;
    private Long receiverId;
    
    @Column(columnDefinition = "TEXT")
    private String content;
    
    private LocalDateTime timestamp;
    private boolean isRead;
    
    // The property context for the chat
    private Long roomId;
    private String roomTitle;

    public ChatMessage() {
        this.timestamp = LocalDateTime.now();
        this.isRead = false;
    }

    public ChatMessage(Long senderId, Long receiverId, String content, Long roomId, String roomTitle) {
        this.senderId = senderId;
        this.receiverId = receiverId;
        this.content = content;
        this.roomId = roomId;
        this.roomTitle = roomTitle;
        this.timestamp = LocalDateTime.now();
        this.isRead = false;
    }
}

File: FRONTEND/shared/src/commonMain/kotlin/org/com/model/ChatMessage.kt
package org.com.model


import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ChatMessage(

    @SerialName("messageId")
    val messageId: Long = 0L,

    @SerialName("senderId")
    val senderId: Long = 0L,

    @SerialName("receiverId")
    val receiverId: Long = 0L,

    @SerialName("content")

    val content: String = "",

    @SerialName("timestamp")
    val timestamp: String = "",

    @SerialName("isRead")
    val isRead: Boolean = false,

    @SerialName("roomId")
    val roomId: Long? = null,

    @SerialName("roomTitle")
    val roomTitle: String? = null
)